modules/pxp-module-puppet

```diff
@@ -111,89 +51,49 @@ def make_error_result(exitcode, error_type, error_message)
-  # root-level Node.
-  data = YAML.parse_file(filename)
-  data.root.each do |o|
-    o.tag = nil if o.respond_to?(:tag=)
-  end
-
-  data.to_ruby
-end
-
-def get_result_from_report(last_run_report, exitcode, config, start_time)
-  if !File.exist?(last_run_report)
-    return make_error_result(exitcode, Errors::NoLastRunReport,
-                             "#{last_run_report} doesn't exist")
-  end
-
-  if start_time && File.mtime(last_run_report) == start_time
-    return make_error_result(exitcode, Errors::NoLastRunReport,
-                             "#{last_run_report} was not written")
-  end
-
-  last_run_report_yaml = {}
-
-  begin
-    last_run_report_yaml = parse_report(last_run_report)
-  rescue => e
-    return make_error_result(exitcode, Errors::InvalidLastRunReport,
-                             "#{last_run_report} could not be loaded: #{e}")
-  end
-
+def get_result_from_report(report, exitcode)
   if exitcode == 0
     run_result = last_run_result(exitcode)
   else
     run_result = make_error_result(exitcode, Errors::NonZeroExit,
                                    "Puppet agent exited with a non 0 exitcode")
   end
 
-  run_result["kind"] = last_run_report_yaml['kind']
-  run_result["time"] = last_run_report_yaml['time']
-  run_result["transaction_uuid"] = last_run_report_yaml['transaction_uuid']
-  run_result["environment"] = last_run_report_yaml['environment']
-  run_result["status"] = last_run_report_yaml['status']
+  run_result["kind"] = report.kind
+  run_result["time"] = report.time
+  run_result["transaction_uuid"] = report.transaction_uuid
+  run_result["environment"] = report.environment
+  run_result["status"] = report.status
 
   return run_result
 end
 
-def start_run(config, action_input)
+def start_run(action_input)
   exitcode = DEFAULT_EXITCODE
-  cmd = make_command_array(config, action_input)
-  env = make_environment_hash()
-
-  last_run_report = check_config_print("lastrunreport", config)
-  if last_run_report.empty?
-    return make_error_result(exitcode, Errors::NoLastRunReport,
-                             "could not find the location of the last run report")
-  end
-  start_time = File.mtime(last_run_report) if File.exist?(last_run_report)
-
-  run_result = Puppet::Util::Execution.execute(cmd, {:failonfail => false,
-                                                     :custom_environment => env,
-                                                     :override_locale => false})
-
-  if !run_result
+  puppet_output = []
+  start_time = nil
+  report = nil
+  begin
+    cmd = Puppet::Util::CommandLine.new('agent', action_input['flags'])
+    Puppet.initialize_settings(cmd.args)
+    agent = Puppet::Agent.new(Puppet::Configurer, false)
+    transaction_uuid = SecureRandom.uuid
+    report = Puppet::Transaction::Report.new("apply", nil, Puppet[:environment], transaction_uuid)
+    exitcode = agent.run(:report => report, :transaction_uuid => transaction_uuid)
+  rescue => e
```

Comment:
i know this was here before, but we really shouldn't be rescuing any exception.  this does all sorts of nasty things like block `Interrupt` and `SignalException` and can result in false passes as it can block beaker's exception handling.